api/sip/__init__.py

```diff
@@ -183,12 +184,20 @@ def __init__(
         self.ssl_key = integration.setting(self.SSL_KEY).value
         self.dialect = Sip2Dialect.load_dialect(integration.setting(self.ILS).value)
         self.client = client
-        patron_status_block = integration.setting(self.PATRON_STATUS_BLOCK).json_value
-        if patron_status_block is None or patron_status_block:
+
+        # Check if patrons should be blocked based on SIP status
+        # If nothing is specified, the default is True (block patrons!)
+        _block = integration.setting(self.PATRON_STATUS_BLOCK).json_value
+        if _block is None:
+            _block = True
+
+        if _block:
+            self.patron_status_should_block = True
```

Comment:
I don't see this variable being used anywhere else

Reply:
OK, this is disturbing. I thought I'd made a mistake and just left in an unused field, but in reality I've actually left out the `if self.patron_status_should_block ...` check later. 

What's disturbing about this? The new unit test shows that with `_block` set to `True`, the excessive fines block happens as it should, and with `_block` set to `False`, the excessive fines block doesn't occur... But we can clearly see that, because I managed to leave out the conditional, the test should fail and the excessive fines should always cause a block. This is into "this can't happen" territory.

Investigating...

Reply:
The `info to block` function checks the block fields, I think thats the reason

Reply:
I believe everything is consistent now.